kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -42,18 +64,59 @@ public VirtualKafkaClusterReconciler(Clock clock) {
 
     @Override
     public UpdateControl<VirtualKafkaCluster> reconcile(VirtualKafkaCluster cluster, Context<VirtualKafkaCluster> context) {
-        var proxyOpt = context.getSecondaryResource(KafkaProxy.class, PROXY_EVENT_SOURCE_NAME);
-        LOGGER.debug("spec.proxyRef.name resolves to: {}", proxyOpt);
+        var existingProxies = context.getSecondaryResource(KafkaProxy.class, PROXY_EVENT_SOURCE_NAME).stream()
```

Comment:
I think this work should be refactored to some new method of `DependencyResolver`, to make it responsible for the similar work of resolving a Proxies dependencies. Could be done as a later PR